objects/src/accounts/account_id.rs

```diff
@@ -0,0 +1,198 @@
+///
+/// Account ID consists of 3 field elements (24 bytes). These field elements uniquely identify a
+/// single account and also specify the type of the underlying account. Specifically:
+/// - If the least significant 32 bits of the 3rd element are all ZEROs, the account is a
+///   fungible asset faucet (i.e., it can issue fungible assets).
+/// - If the least significant 32 bits of the 3rd element are set to 2^31 (i.e ONE followed by 31
+///   ZEROs), the account is a non-fungible asset faucet (i.e., it can issue non-fungible assets).
+/// - Otherwise, the account is a regular account.
+///
+/// Additionally, account IDs have the following properties
+/// - For fungible asset faucets account IDs are guaranteed to start with ONE.
+/// - For regular accounts, the last 3 bytes of the ID are guaranteed to be all ZEROs.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct AccountId(Felt, Felt, Felt);
+
+impl AccountId {
+    // CONSTANTS
+    // --------------------------------------------------------------------------------------------
+    pub const FUNGIBLE_FAUCET_TAG: u32 = 0;
+    pub const NON_FUNGIBLE_FAUCET_TAG: u32 = 1 << 31;
+
+    pub const MIN_TRAILING_ZEROS: u32 = 24;
+
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    /// Returns a new account ID derived from the specified seed.
+    ///
+    /// The account ID is computed by hashing the seed and using elements 0, 1, and 3 of the result
+    /// (i.e., omitting element 2) for the account ID.
+    ///
+    /// # Errors
+    /// Returns an error if the resulting account ID does not comply with account ID rules.
+    pub fn new(seed: Word) -> Result<Self, AccountError> {
+        // hash the seed and build the ID from the 1st, 2nd, and 4th elements of the result
+        let hash = Hasher::hash_elements(&seed);
+        let id = Self(hash[0], hash[1], hash[3]);
+
+        // verify that the ID satisfies all rules
+        id.validate()?;
+
+        Ok(id)
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns true if an account with this ID can issue fungible assets.
+    pub fn is_fungible_faucet(&self) -> bool {
+        self.tag() == Self::FUNGIBLE_FAUCET_TAG
+    }
+
+    /// Returns true if an account with this ID can issue non-fungible assets.
+    pub fn is_non_fungible_faucet(&self) -> bool {
+        self.tag() == Self::NON_FUNGIBLE_FAUCET_TAG
+    }
+
+    /// Returns true if an account with this ID can issue assets.
+    pub fn is_faucet(&self) -> bool {
+        self.is_fungible_faucet() || self.is_non_fungible_faucet()
+    }
+
+    // SEED GENERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Finds and returns a seed suitable for creating regular account IDs using the provided seed
+    /// as a string point.
```

Comment:
nit: I think this should be "...as a starting point". If so, it should be updated for the functions below as well.